dali/python/backend_impl.cc

```diff
@@ -195,8 +195,51 @@ void CheckContiguousTensor(const TStrides &strides, const TShape &shape, size_t
   CheckContiguousTensor(strides, dali::size(strides), shape, dali::size(shape), element_size);
 }
 
+template <typename Backend>
+void SetLayout(
+      Tensor<Backend> &t,
+      const std::optional<std::string> &layout_str,
+      bool clear_if_none = true) {
+  if (layout_str && !layout_str->empty()) {
+    TensorLayout layout = *layout_str;
+    if (t.ndim() != layout.ndim()) {
+      throw py::value_error(make_string(
+        "A non-empty layout must have the same number of dimensions as the "
+        "number of dimensions of the Tensor.\n"
+        "Got: ", layout.ndim(), " (", layout, ")\n",
+        "Expected: ", t.ndim(), "."));
+    }
+    t.SetLayout(layout);
+  } else if (clear_if_none) {
+    t.SetLayout({});
+  }
+}
+
+template <typename Backend>
+void SetLayout(
```

Comment:
Yeah, I know it's almost the same - as the one above - but the message is different and the API is different (ndim vs sample_dim).